logs-lib/utils.libsonnet

```
@@ -0,0 +1,37 @@
+{
+  local this = self,
+
+  labelsToURLvars(labels, prefix)::
+    std.join('&', ['var-%s=${%s%s}' % [label, prefix, label] for label in labels]),
+
+  // For PromQL or LogQL
+  labelsToPromQLSelector(labels): std.join(',', ['%s=~"$%s"' % [label, label] for label in labels]),
+  labelsToLogQLSelector: self.labelsToPromQLSelector,
+
+  labelsToPanelLegend(labels): std.join('/', ['{{%s}}' % [label] for label in labels]),
+
+  // Generate a chain of labels. Useful to create chained variables
+  chainLabels(labels, additionalFilters=[]):
+    local last(arr) = std.reverse(arr)[0];
+    local chainSelector(chain) =
+      std.join(
+        ',',
+        if std.length(chain) > 0
+        then additionalFilters + [this.labelsToPromQLSelector(chain)]
+        else additionalFilters
```

I'd reduce the scope of the conditional as much as possible, the length of the chain isn't relevant to `additionalFilters`.
```suggestion
        additionalFilters 
        + (if std.length(chain) > 0
            then [this.labelsToPromQLSelector(chain)]
            else [])
```